sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/IDigitalTwinRelationship.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.digitaltwins.core;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Interface that defines the Json properties that are present in every digital twin relationship. Implementations of
+ * this class must add Json properties for each of the getters defined in this interface. See each getter for more
+ * details on each Json property.
+ */
+public interface IDigitalTwinRelationship {
+    public static String DIGITAL_TWIN_RELATIONSHIP_ID_JSON_PROPERTY_NAME = "$relationshipId";
+    public static String DIGITAL_TWIN_RELATIONSHIP_SOURCE_ID_JSON_PROPERTY_NAME = "$sourceId";
+    public static String DIGITAL_TWIN_RELATIONSHIP_TARGET_ID_JSON_PROPERTY_NAME = "$targetId";
+    public static String DIGITAL_TWIN_RELATIONSHIP_NAME_JSON_PROPERTY_NAME = "$relationshipName";
+
+    /**
+     * Gets the unique Id of the relationship. This field is present on every relationship.
+     * <p>
+     * The Json property associated with this Id is {@link IDigitalTwinRelationship#DIGITAL_TWIN_RELATIONSHIP_ID_JSON_PROPERTY_NAME},
+     * so every implementation of this class must have a Json property with that value. As an example, see
+     * {@link com.azure.digitaltwins.core.models.BasicRelationship} which demonstrates how to create this
+     * property using {@link JsonProperty}.
+     *
+     * @return The unique Id of the relationship. This field is present on every relationship.
+     */
+    public String getRelationshipId();
```

Comment:
In .NET we just called this Id, as the fact it is a relationship is clear (I think anyway) from the type. Do you think it is necessary to call it relationshipId?

Reply:
This was a comment we got during API review. The fact that there are 2 other "Id" fields in here made this one feel like a weak name when it was just ```getId()```